sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py

```diff
@@ -254,11 +255,14 @@ def get_queue_receiver(self, queue_name, **kwargs):
                 "To connect to the sub queue of a sessionful queue, "
                 "please set sub_queue only as sub_queue does not support session."
             )
-        if sub_queue and sub_queue in SubQueue:
+        try:
             queue_name = generate_dead_letter_entity_name(
                 queue_name=queue_name,
-                transfer_deadletter=(sub_queue == SubQueue.TransferDeadLetter)
+                transfer_deadletter=(ServiceBusSubQueue(sub_queue) == ServiceBusSubQueue.TransferDeadLetter)
```

Comment:
This works if `sub_queue` is both str and an enum instance? I think it does IIRC - but just double checking

Reply:
Correct.